test/Authorization.t.sol

```diff
@@ -51,6 +51,107 @@ contract SparkVaultSetSsrBoundsSuccessTests is SparkVaultTestBase {
 
 }
 
+contract SparkVaultRoleManagementFailureTests is SparkVaultTestBase {
+
+    function test_grantRole_notAdmin() public {
+        bytes32[] memory roles = new bytes32[](3);
+        roles[0] = DEFAULT_ADMIN_ROLE;
+        roles[1] = SETTER_ROLE;
+        roles[2] = TAKER_ROLE;
+
+        for (uint256 i = 0; i < roles.length; i++) {
+            bytes32 role = roles[i];
+            vm.expectRevert(abi.encodeWithSignature(
+                "AccessControlUnauthorizedAccount(address,bytes32)",
+                address(this),
+                DEFAULT_ADMIN_ROLE
+            ));
+            vault.grantRole(role, address(0x1234));
+        }
+    }
+
+    function test_revokeRole_notAdmin() public {
+        bytes32[] memory roles = new bytes32[](3);
+        roles[0] = DEFAULT_ADMIN_ROLE;
+        roles[1] = SETTER_ROLE;
+        roles[2] = TAKER_ROLE;
+
+        for (uint256 i = 0; i < roles.length; i++) {
+            bytes32 role = roles[i];
+            vm.expectRevert(abi.encodeWithSignature(
+                "AccessControlUnauthorizedAccount(address,bytes32)",
+                address(this),
+                DEFAULT_ADMIN_ROLE
+            ));
+            vault.revokeRole(role, address(0x1234));
+        }
+    }
+
+}
+
+contract SparkVaultRoleManagementSuccessTests is SparkVaultTestBase {
+
+    event RoleGranted(bytes32 indexed role, address indexed account, address indexed sender);
+    event RoleRevoked(bytes32 indexed role, address indexed account, address indexed sender);
+
+    function test_grantRole() public {
+        bytes32[] memory roles = new bytes32[](3);
+        roles[0] = DEFAULT_ADMIN_ROLE;
+        roles[1] = SETTER_ROLE;
+        roles[2] = TAKER_ROLE;
+
+        // admin (DEFAULT_ADMIN_ROLE) should be allowed to grant DEFAULT_ADMIN_ROLE, SETTER_ROLE,
+        // TAKER_ROLE.
+        vm.startPrank(admin);
+        for (uint256 i = 0; i < roles.length; i++) {
+            bytes32 role = roles[i];
+            assertFalse(vault.hasRole(role, address(0x1234)), "Role already granted");
```

Comment:
We don't use error messages in tests unless theyre in a helper function. They become a lot to manage and the assertions are easy enough to understand/locate on a reversion